kratos/includes/constitutive_law.h

```diff
@@ -82,6 +82,11 @@ class KRATOS_API(KRATOS_CORE) ConstitutiveLaw : public Flags
     typedef std::size_t SizeType;
     typedef Geometry<Node < 3 > > GeometryType;
 
+    typedef array_1d<double, 6> StrainVectorType;
```

Comment:
```suggestion
    typedef array_1d<double, 6> StrainVector3DType;
```

Reply:
unfortunately array_1d is actually of fixed size. We need to use "BoundedVector" to be resizeable.

in any case i would propose to do this change in two steps:
1. add the typedefs but with "Vector" and "Matrix" (so still heap array) and change the code so that in the CL we never use the Vector and Matrix but rather always the typedefs
2. actually change the typedefs to the BoundedVector/Matrix

Reply:
Indeed, and considering that this PR will be a pain in the a**, I'd follow @RiccardoRossi 's  apprach but doing 2 PRs, one for step 1 and one for step 2.
In PR 1 I'd also change the function signatures as @loumalouomega suggests below. Even though I'm afraid that this will not raise any compilation error, so we will miss some of them, I think will make things easier for 2nd PR.

Reply:
ok... I'll make in this pr then the typedef's with the current `Vector `and `MAtrix` but changing all the stuff in the element and CL